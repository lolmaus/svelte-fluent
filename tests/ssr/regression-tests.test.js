import { describe, it, expect, vi } from 'vitest';

vi.stubGlobal('navigator', { languages: [] });

const testComponents = import.meta.glob('../../src/regression-tests/**/Test.svelte');

export const tests = Object.entries(testComponents).map(([name, mod]) => ({
	name: name.replace('../../src/regression-tests/', ''),
	module: mod
}));

for (const test of tests) {
	describe(`Regression test ${test.name}`, () => {
		it('should render', async () => {
			const component = (await test.module()).default;
			const res = component.render();
			expect(res).toMatchSnapshot();
		});
	});
}
