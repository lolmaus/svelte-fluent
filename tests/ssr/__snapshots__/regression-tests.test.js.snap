// Vitest Snapshot v1

exports[`Regression test empty-pattern-with-attributes/Test.svelte > should render 1`] = `
{
  "css": {
    "code": "",
    "map": null,
  },
  "head": "",
  "html": "Some text",
}
`;
